radix-engine/src/vm/wasm_runtime/scrypto_runtime.rs

```diff
@@ -371,10 +374,10 @@ where
             .consume_cost_units(ClientCostingEntry::RunWasmCode {
```

Comment:
This code took me longer than I care to admit to understand and validate. Could we update the comment to explain that we're taking:
```
/// If we need to request more from the fee reserve, we round `n` up to the nearest `1_000_000` and request
/// that much additional gas
let amount_to_request = ...
```